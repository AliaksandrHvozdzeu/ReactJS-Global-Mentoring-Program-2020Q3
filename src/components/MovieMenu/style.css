.movie-menu-button {
    position: absolute;
    height: 5px;
    width: 5px;
    top: 75%;
    left: 50%;
    border-radius: 50%;
    transform: translateX(-50%) translateY(-50%);
    background-color: var(--white);
}

.movie__menu {
    position: absolute;
    top: 1rem;
    right: 1rem;
    background-color: var(--grey-darker-transparent);
    border-radius: 50%;
    padding: 0;
    margin: 0;
    cursor: pointer;
}

.movie-menu-closed {
    width: 40px;
    height: 40px;
    display: none;
}

.movie-menu-closed:hover {
    background-color: var(--grey-transparent);
}

.movie-menu-closed:active {
    background-color: var(--grey-darker);
}

.movie-menu-closed::before {
    content: '';
    position: absolute;
    height: 5px;
    width: 5px;
    top: 25%;
    left: 50%;
    border-radius: 50%;
    transform: translateX(-50%) translateY(-50%);
    background-color: var(--white);
}

.movie-menu-closed::after {
    content: '';
    position: absolute;
    height: 5px;
    width: 5px;
    top: 50%;
    left: 50%;
    border-radius: 50%;
    transform: translateX(-50%) translateY(-50%);
    background-color: var(--white);
}

.movie-menu-opened {
    border-radius: 5px;
    width: 10rem;
    height: 7rem;
    padding: 1.5rem 0 1rem;
    cursor: default;
}

.movie-menu-action {
    width: 100%;
    border-radius: 0;
    background-color: transparent;
    text-align: left;
    color: var(--white);
    display: block;
    font-size: 1.2rem;
    font-weight: 200;
    padding: 0.5rem 1rem;
    outline: none;
}

.movie-menu-action:hover {
    background-color: var(--red);
}

.movie-menu-close {
    position: absolute;
    right: 0;
    top: 0;
    background-color: transparent;
    color: transparent;
    border: transparent;
    outline: none;
}

.movie-menu-close::before {
    content: '';
    position: absolute;
    border-top: 2px solid var(--white);
    transform: rotate(45deg);
    width: 1rem;
    left: 30%;
    top: 80%;
}

.movie-menu-close::after {
    content: '';
    position: absolute;
    border-top: 2px solid var(--white);
    transform: rotate(-45deg);
    width: 1rem;
    left: 30%;
    top: 80%;
}
