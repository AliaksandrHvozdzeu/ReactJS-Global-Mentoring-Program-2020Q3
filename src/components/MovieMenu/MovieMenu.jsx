import React, { useState } from 'react';
import PropTypes from 'prop-types';
import './style.css';


const MenuClosed = ({ action }) => (
  <button type="button" className="movie-menu movie-menu-closed" onClick={() => action(true)}>
    <div className="movie-menu-button"/>
  </button>
);

const MenuOpened = ({ action, editMovie, deleteMovie }) => (
  <div className="movie-menu movie-menu-opened">
    <button className="movie-menu-action" type="button" onClick={editMovie}>Edit</button>
    <button className="movie-menu-action" type="button" onClick={deleteMovie}>Delete</button>
    <button className="movie-menu-close" id="movie-menu-close-button" type="button" onClick={() => action(false)}>Close</button>
  </div>
);

MenuClosed.propTypes = {
  action: PropTypes.func.isRequired
};

MenuOpened.propTypes = {
  action: PropTypes.func.isRequired,
  editMovie: PropTypes.func.isRequired,
  deleteMovie: PropTypes.func.isRequired
};

const MovieMenu = ({ editAction, deleteAction }) => {
  const [isOpened, setIsOpened] = useState(false);
  return (
    <>
      {!isOpened && <MenuClosed action={setIsOpened}/>}
      {isOpened && (<MenuOpened action={setIsOpened} editMovie={editAction} deleteMovie={deleteAction}/>)}
    </>
  );
};

MovieMenu.propTypes = {
  editAction: PropTypes.func.isRequired,
  deleteAction: PropTypes.func.isRequired
};

export default MovieMenu;
