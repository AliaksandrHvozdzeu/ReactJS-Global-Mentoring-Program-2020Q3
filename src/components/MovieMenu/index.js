import MovieMenu from './js/MovieMenu';

export default MovieMenu;
