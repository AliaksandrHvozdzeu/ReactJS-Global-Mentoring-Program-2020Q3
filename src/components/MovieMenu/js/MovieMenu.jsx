import React, { useState } from "react";
import PropTypes from "prop-types";
import "../css/MovieMenu.css";
import MovieMenuClose from "./MovieMenuClose";
import MovieMenuOpen from "./MovieMenuOpen";

const MovieMenu = ({ editAction, deleteAction }) => {
  const [isOpened, setIsOpened] = useState(false);
  return (
    <>
      {!isOpened && <MovieMenuClose action={setIsOpened} />}
      {isOpened && (<MovieMenuOpen action={setIsOpened}
                                   editMovie={editAction}
                                   deleteMovie={deleteAction} />)}
    </>
  );
};

MovieMenu.propTypes = {
  editAction: PropTypes.func.isRequired,
  deleteAction: PropTypes.func.isRequired
};

export default MovieMenu;
