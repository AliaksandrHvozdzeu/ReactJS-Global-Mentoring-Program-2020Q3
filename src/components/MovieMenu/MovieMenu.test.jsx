import React from "react";
import rerender from "react-test-renderer";
import MovieMenu from "./js/MovieMenu";

describe("when CloseButton", () => {

  test("then snapshot created", () => {
    const editAction = () => {
    };
    const deleteAction = () => {
    };
    const component = rerender.create(<MovieMenu editAction={editAction}
                                                 deleteAction={deleteAction} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});
