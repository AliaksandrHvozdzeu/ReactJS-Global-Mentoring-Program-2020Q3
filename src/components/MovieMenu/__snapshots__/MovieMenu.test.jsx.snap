// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when CloseButton then snapshot created 1`] = `
<button
  className="movie-menu movie-menu-closed"
  onClick={[Function]}
  type="button"
>
  <div
    className="movie-menu-button"
  />
</button>
`;
